src/PostgREST/DbStructure.hs

```diff
@@ -311,22 +313,40 @@ accessibleTables =
       relname as table_name,
       d.description as table_description,
       (
-        c.relkind in ('r', 'v', 'f')
-        and (pg_relation_is_updatable(c.oid::regclass, false) & 8) = 8
-        -- The function `pg_relation_is_updateable` returns a bitmask where 8
-        -- corresponds to `1 << CMD_INSERT` in the PostgreSQL source code, i.e.
-        -- it's possible to insert into the relation.
-        or (exists (
-          select 1
-          from pg_trigger
-          where
+        c.relkind IN ('r', 'v','f')
+        AND (pg_relation_is_updatable(c.oid::regclass, FALSE) & 8) = 8
+        OR EXISTS (
+          SELECT 1
+          FROM pg_trigger
+          WHERE
             pg_trigger.tgrelid = c.oid
-            and (pg_trigger.tgtype::integer & 69) = 69)
-            -- The trigger type `tgtype` is a bitmask where 69 corresponds to
-            -- TRIGGER_TYPE_ROW + TRIGGER_TYPE_INSTEAD + TRIGGER_TYPE_INSERT
-            -- in the PostgreSQL source code.
+            AND (pg_trigger.tgtype::integer & 69) = 69
         )
-      ) as insertable
+      ) AS insertable,
+      (
+        c.relkind IN ('r', 'v','f')
+        AND (pg_relation_is_updatable(c.oid::regclass, FALSE) & 4) = 4
+        -- CMD_UPDATE
+        OR EXISTS (
+          SELECT 1
+          FROM pg_trigger
+          WHERE
+            pg_trigger.tgrelid = c.oid
+            and (pg_trigger.tgtype::integer & 81) = 81
+        )
+      ) as updatable,
+      (
+        c.relkind IN ('r', 'v','f')
+        AND (pg_relation_is_updatable(c.oid::regclass, FALSE) & 16) = 16
```

Comment:
These `c.relkind` and `pg_relation_is_updatable` parts are repeated in all the subqueries. Would it be possible to put them in a CTE?

Reply:
How would you do that? They all compare against a different bitmask.

Reply:
> They all compare against a different bitmask.

I also think it wouldn't be possible in this case. The `c.relkind` on the otherhand may be, but I don't know if it's worth the use of a CTE only for a line of code.

On a side note, when looking at the [source code,](https://doxygen.postgresql.org/rewriteHandler_8c.html#a52a3ea381d9db1b6e917b7280e3e6d71) postgres treats ordinary tables, that is `c.relkind = 'r'`, as always updatable with all events. Maybe it could be better to verify first if it's an ordinary table and use `pg_relation_is_updatable` only for views, something like [this](https://github.com/postgres/postgres/blob/master/src/backend/catalog/information_schema.sql#L1989-L1993).